### PR TITLE
bksahu's gsoc wrap up added

### PR DESCRIPTION
**Pull request by bksahu**

### CODE REVIEW

File: posts/bksahus-gsoc-wrap-up.rst [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+* Issue reported and closed `Issue #349 <https://github.com/Nuitka/Nuitka/issues/349>`
+    * Created a new module `nuitka.nodes.IterationHandles` to work with iterables.
+    * Added support of Iteration for non-mutable types.
+
+* "all": `PR #407 <https://github.com/Nuitka/Nuitka/pull/407>`
+    * Added `nuitka.nodes.BuiltinAllNodes` to optimize the "all" built-ins.
+    * Developed an algorithm similar to "any" to predict the "all" arguments.
+      For example::
+      
+        all([0, 0, 1]) -> False
```
Comment (kayhayen) [inline posts/bksahus-gsoc-wrap-up.rst]: Is this not missign a `code-block` directive?

File: posts/bksahus-gsoc-wrap-up.rst [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+=============
+* Nuitka git work flow: `PR #485 https://github.com/Nuitka/Nuitka/pull/485`
+
+And other minor doc fixes are added with their respective pull requests.
+
+What I learned
+==============
+* Learned the software engineering principles and how to keep my work clean.
+* I also learned how to effectively use software designing principles like DRY and KISS.
+* Got exposed to Nuikta internals which helped me to better understand how compilers in general work.
```
Comment (kayhayen) [inline posts/bksahus-gsoc-wrap-up.rst]: Typo of very important word :)

File: posts/bksahus-gsoc-wrap-up.rst (hunk 1 above)
Comment (bksahu) [reply posts/bksahus-gsoc-wrap-up.rst]: It's already in a `code-block` directive

File: posts/bksahus-gsoc-wrap-up.rst [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+   Pseudo-code of "zip" reformulation::
+     
+       def _zip(a, b, c, ... ):
+       # First assign, to preserve order of execution,
+       # the arguments might be complex expressions.
+       tmp_arg1 = a
+       tmp_arg2 = b
+       tmp_arg3 = c
+       ...
+        tmp_iter_1 = iter(tmp_arg1)
```
Comment (kayhayen) [inline posts/bksahus-gsoc-wrap-up.rst]: Identation mistake there is a space before `tmp_iter_1 =` that should not be there.

File: posts/bksahus-gsoc-wrap-up.rst [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+            ...
+         return result
+
+   * Adding support for `keyfunc` is pending
+
+* "zip": `PR #462 <https://github.com/Nuitka/Nuitka/pull/462>`
+   * This built-in uses both types of optimizations that the previous built-ins optimizations used.
+   * `zip` for Python 2 uses the reformulations.
+   Pseudo-code of "zip" reformulation::
+
```
Comment (kayhayen) [inline posts/bksahus-gsoc-wrap-up.rst]: `.. code-block:: python ` is not the same as `::` to Nikola.

File: posts/bksahus-gsoc-wrap-up.rst [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+==============
+* Learned the software engineering principles and how to keep my work clean.
+* I also learned how to effectively use software designing principles like DRY and KISS.
+* Got exposed to Nuitka internals which helped me to better understand how compilers in general work.
+* Explored how CPython works internally.
+* Got some great advice from Mentor Hayen about starting my professional career in Software engineering.
+
+Overall, it was a great experience to be a part of Nuitka :) 
+
+Yours, :raw-html:`<br />`
```
Comment (kayhayen) [inline posts/bksahus-gsoc-wrap-up.rst]: Out of curiousity, I looked up, how to do it.

Check out http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html and "line blocks".

I think the way to do it is this then:
```
| Yours,
| Kay
```

I will try this out on my existing posts eventually (works this way). But you can leave it like this, I am not yet sure if links work in there and it's only a style/knowledge issue.

I like btw how you made your name a link.

**Comment by kayhayen** [comment]
Just to annoy you no further, I am merging now, but these some of those blocks actually were block quotes and definitely not shell, and only some were Python. I will fix it up myself though, no big deal.

**Pull request merged**